### PR TITLE
#13246: Remove unary_backward_op.hpp

### LINKED ISSUE #13246: Remove `unary_backward_op.hpp`
Work : 

- Remove `ttnn/cpp/ttnn/operations/eltwise/unary_backward/device/unary_backward_op.hpp` 
- Delete `ttnn/cpp/ttnn/operations/eltwise/unary_backward/device` 
- Move `ttnn/cpp/ttnn/operations/eltwise/unary_backward/device/unary_backward_op.cpp` to `ttnn/cpp/ttnn/operations/eltwise/unary_backward/unary_backward.cpp`

### PR DESCRIPTION
**Pull request by VirdhatchaniKN**
Issue : #13246

[CI](https://github.com/tenstorrent/tt-metal/actions/runs/11110152125) : PASSED

### CODE REVIEW

**Review by Aswinmcw — APPROVED** [review]
LGTM

File: ttnn/cpp/ttnn/operations/eltwise/unary_backward/unary_backward.hpp [hunk 1]
```diff
@@ -5,14 +5,15 @@
...
 
-#include "device/unary_backward_op.hpp"
 #include "ttnn/device_operation.hpp"
 #include "ttnn/operations/eltwise/complex/complex.hpp"
 
 namespace ttnn {
 
 namespace operations::unary_backward {
 
+Tensor change_layout_to_tile(const Tensor& temp, const MemoryConfig& output_mem_config);
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/operations/eltwise/unary_backward/unary_backward.hpp:15]: new declaration added but I can't find the definition

Comment (VirdhatchaniKN) [reply ttnn/cpp/ttnn/operations/eltwise/unary_backward/unary_backward.hpp:15]: It is present in the cpp file - [Link to File](https://github.com/tenstorrent/tt-metal/blob/142e4ab41cda0d782c0c602c266a1182e4979e0a/ttnn/cpp/ttnn/operations/eltwise/unary_backward/unary_backward.cpp#L1454)

**Review by eyonland — APPROVED** [review]

**Review by ayerofieiev-tt — APPROVED** [review]

**Review by ntarafdar — APPROVED** [review]

**Review by sjameelTT — APPROVED** [review]

**Comment by VirdhatchaniKN** [comment]
https://github.com/tenstorrent/tt-metal/actions/runs/11137666399 - passed

**Pull request merged**